src/mappings/handlers.ts

```diff
@@ -0,0 +1,278 @@
+import { BigInt, ByteArray, crypto } from '@graphprotocol/graph-ts'
+import { BridgeTransaction } from '../../generated/schema'
+import { TokenDeposit, TokenDepositAndSwap, TokenMint, TokenMintAndSwap, TokenRedeem, TokenRedeemAndRemove, TokenRedeemAndSwap, TokenWithdraw, TokenWithdrawAndRemove } from '../../generated/SynapseBridge/SynapseBridge'
+import { TokenDefinition } from './tokens'
+import { handleOut } from './utils'
+
+// IN
+export function handleTokenMintAndSwap(event: TokenMintAndSwap): void {
+    const blockNumber = event.block.number
+    const timestamp = event.block.timestamp
+    const txnHash = event.transaction.hash
+    const kappa = crypto.keccak256(ByteArray.fromHexString(txnHash.toHexString())).toHexString()
+    
+    let toAddress = event.params.to
+    let fee = event.params.fee
+    let swapSuccess = event.params.swapSuccess
+    let tokenIndexTo = event.params.tokenIndexTo
+    let receivedTokenAddress = event.params.token // todo: indexer has other logic
+    let receivedTokenSymbol = TokenDefinition.fromAddress(receivedTokenAddress).symbol
+    let receivedTokenValue = event.params.amount
+    let toChainId = BigInt.fromI32(7700) // todo
+    let pending = false
+
+    if (!swapSuccess) {
+        receivedTokenValue = receivedTokenValue.minus(fee)
+    }
+
+    // update bridge txn
+    let txn = BridgeTransaction.load(kappa)
+    if (txn == null) {
+        txn = new BridgeTransaction(kappa)
+    }
+
+    if (txn.toTxnHash) {
+        pending = false
+    }
+
+    // update object
+    // txn.toTxnHash = null 
+    txn.kappa = kappa
```

Comment:
there seems to be a lot of code duplication in the `in` functions. Can you create a function handleIn() similar to `handleOut()` which contains all the common stuff which is getting duplicated

Reply:
interfaces and classes are not compiling using `graph build` thats why i chose to impl them in each function

Reply:
what is the error while compiling the interfaces and classes? There is no way to fix that error?

Also `handleIn()` is present, so `handleOut()` should be quite similar right? (we want `handleOut()` only for the duplicated part, rest can remain as it is)